### PR TITLE
feat: Add a global flag `--wait-for-sync`

### PR DESCRIPTION
**Pull request by keroro520**
https://github.com/nervosnetwork/ckb-cli/issues/199

`--wait-for-sync` is a global flag, indicates a requirement that the command must not be executing unless the index-store synchronize to tip before executing the command.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -302,3 +327,32 @@ pub fn build_interactive() -> App<'static, 'static> {
...
+            IndexDatabase::from_db(backend, cf, network_type, genesis_info.clone(), false)
+                .map(|db| db.last_number().unwrap_or(0))
+                .or_else(|_| Ok(0))
+        });
+        if synced.unwrap_or(0) == tip_number {
+            break;
+        }
+    }
+    Ok(())
+}
```
Comment (TheWaWaR) [inline src/main.rs]: Put this function in `src/utils/other.rs`

**Comment by TheWaWaR** [comment]
When `--wait-for-sync` is present, interactive only limitation should be removed.

File: src/main.rs [hunk 2]
```diff
@@ -89,6 +95,17 @@ fn main() -> Result<(), io::Error> {
...
+
+    {
+        let wait_for_sync = matches.is_present("wait-for-sync");
+        if wait_for_sync {
+            if let Err(err) = sync_to_tip(&mut rpc_client, &index_dir) {
+                eprintln!("Synchronize error: {}", err);
+                process::exit(1);
+            }
+        }
+    }
```
Comment (TheWaWaR) [inline src/main.rs]: Brace is not necessary, you may add some comment above the code.

**Comment by keroro520** [comment]
> When `--wait-for-sync` is present, interactive only limitation should be removed.

Updated.

**Pull request merged**